### ISSUE TITLE
[Server] Implement Polls and Live Voting feature API route

### ISSUE THREAD

**Issue by shubhansu-kr** [mvp-2003/Quikz#22]
##  Polls & Live Voting
**Description:**  
Implement real-time polling and voting functionality for users, similar to Mentimeter.

**Tasks:**
- Create poll model.
- Implement routes to create, manage, and participate in polls.
- Implement real-time voting logic.
- Add routes to fetch live poll results.

**Dependencies:**  
- **User Management** (Issue 1) for authentication.
- **Quiz Management** (Issue 2) for integrating quizzes with polls.

**Guidelines for Assignee:**
- Comment on the issue with the model and routes you will implement.
- Follow the quiz routes structure to maintain consistency.

**Comment by mohitahlawat2001** [comment]
I want to work on this please assign it to me

**Comment by shubhansu-kr** [comment]
Hi @mohitahlawat2001 , You can start working on this. Please share your approach here and then implement it. We would appreciiate it if you document your work for future reference too. Thanks

**Comment by mohitahlawat2001** [comment]
- POST /polls: create a new poll
- GET /polls: retrieve a list of all polls
- GET /polls/:id: retrieve a specific poll by ID
- PUT /polls/:id: update a specific poll by ID
- DELETE /polls/:id: delete a specific poll by ID
- POST /polls/:id/votes: cast a vote for a specific poll option
- GET /polls/:id/results: retrieve the live poll results


schema like this ,
```
title: String,
  description: String,
  options: [
    {
      option: String,
      votes: { type: Number, default: 0 }
    }
  ],
  created_at: { type: Date, default: Date.now },
  updated_at: { type: Date, default: Date.now }
```

for realtime can use sockets

**Comment by shubhansu-kr** [comment]
@mohitahlawat2001 How do you plan on handling the case where we might have multiple question in the poll

**Comment by mohitahlawat2001** [comment]
can make object for question

  question: String,
  options: [
    {
      option: String,
      votes: { type: Number, default: 0 }
    }
  ]
});

**Comment by shubhansu-kr** [comment]
Sounds Right!

**Comment by mohitahlawat2001** [comment]
should i start working on this

**Comment by shubhansu-kr** [comment]
Closed by #44
